File: FreeIceCream.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int n, x;
    cin >> n >> x;
    long long iceCream=x, kids=0;
    for (int i=0; i<n; i++) {
        char sign;
        cin >> sign;
        long long ic;
        cin >> ic;
        if (sign=='+') 
            iceCream += ic;
        else if (sign=='-' && iceCream>=ic)
            iceCream -= ic;
        else if (sign=='-' && iceCream<ic) 
            kids++;
    }
    cout << iceCream << " " << kids << endl;

    return 0;
}

File: Python/FreeIceCream.py
def main():
    n = int(input())
    x = int(input())
    iceCream = x
    kids = 0
    for i in range(n):
        sign = input()
        ic = int(input())
        if sign == "x":
            iceCream += ic
        elif sign == '-' and iceCream >= ic:
            iceCream -= ic
        elif sign == '-' and iceCream < ic:
            kids += 1
    print(f"{iceCream} {kids}")


if __name__ == '__main__':
    main()
